pdr_backend/predictoor/predictoor_agent.py

```diff
@@ -481,22 +482,23 @@ def get_payout(self):
         pending_slots = query_pending_payouts(
             self.ppss.web3_pp.subgraph_url, up_pred_addr
         )
-        contracts = list(pending_slots.keys())
-        contracts_checksummed = [
-            self.ppss.web3_pp.web3_config.w3.to_checksum_address(addr)
-            for addr in contracts
-        ]
-        slots = list(pending_slots.values())
-        slots_flat = [item for sublist in slots for item in sublist]
-        slots_unique_flat = list(set(slots_flat))
-        if len(slots_unique_flat) == 0:
+        shared_slots = find_shared_slots(pending_slots)
+        if len(shared_slots) == 0:
             logger.info("No payouts available")
             return
 
-        tx = self.pred_submitter_mgr.get_payout(
-            slots_unique_flat, contracts_checksummed
-        )
-        logger.info("Payout tx: %s", tx["transactionHash"].hex())
+        for slot_tuple in shared_slots:
+            contract_addresses = slot_tuple[0]
+            contracts_checksummed = [
```

Comment:
Rename `contracts_checksummed` -> `contract_addrs`, in all locations in this method. 

Why: `contracts_checksummed` is not contracts per se, but contract addresses. The variable name should reflect this. And going through a checksum filter doesn't make it worth the extra complexity of the variable name to include `checksummed`.

----
And in fact the whole 4 lines starting with `contracts_checksummed` can be converted to a single line using self's `_to_checksum()` method.

That is, from this:

```python
            contracts_checksummed = [
                self.ppss.web3_pp.web3_config.w3.to_checksum_address(addr)
                for addr in contract_addresses
            ]
```

To this:
```python
contract_addrs = self._to_checksum(contract_addrs)
```